app/demo-loop/LDemoIO.root.cc

```diff
@@ -0,0 +1,66 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2022 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file demo-loop/LDemoIO.root.cc
+//---------------------------------------------------------------------------//
+#include "LDemoIO.hh"
+
+#include <TBranch.h>
+#include <TFile.h>
+#include <TTree.h>
+
+namespace demo_loop
+{
+//---------------------------------------------------------------------------//
+void to_root(std::shared_ptr<celeritas::RootFileManager> root_manager,
+             LDemoArgs&                                  args)
+{
+    CELER_ASSERT(args);
+    CELER_ASSERT(root_manager);
+
+    // So far RootFileManager has only a single TFile instance and ROOT knows
+    // that this new TTree has to be attached to this single TFile. If we
+    // expand it to a RootFileManager::tfile_[tid] we'd need to specify the
+    // correct TFile* by invoking TTree("name", "title", TFile*).
+    std::unique_ptr<TTree> tree_input
+        = std::make_unique<TTree>("input", "input");
+
+    // Problem definition
+    tree_input->Branch("geometry_filename", &args.geometry_filename);
+    tree_input->Branch("physics_filename", &args.physics_filename);
+    tree_input->Branch("hepmc3_filename", &args.hepmc3_filename);
+
+    // Control
+    tree_input->Branch("seed", &args.seed);
+    // {
+    // Celeritas' type aliases have to be manually casted to be stored properly
+    // See leaflist types: https://root.cern.ch/doc/master/classTBranch.html
+    tree_input->Branch(
+        "max_num_tracks", &args.max_num_tracks, "max_num_tracks/l");
+    tree_input->Branch("max_steps", &args.max_steps, "max_num_steps/l");
```

Comment:
Yeah, I thought about that. We can make a "manual" decision based on `sizeof(size_type)`, but my thought is that since the root output is always 64 bits, we should always still get the correct number, since the cast would either be 64 -> 64 or 32 -> 64. Am I missing something?

Reply:
```suggestion
    tree_input->Branch(
        "max_num_tracks", &args.max_num_tracks);
    tree_input->Branch("max_steps", &args.max_steps);
```

Reply:
And same every where.  I.e. let the compiler discover the type.

Reply:
@pcanal I tried that. Somehow ROOT doesn't get the MACRO defined types (real_type, size_type), that's why I manually casted them. If you look around those branches, the others are just using the references and letting ROOT figure them out.

Reply:
That's does not sound right.  By the time the template 'see' the type all the MACRO and typedefs have been resolved, so there is no reason why it would not work (it may not be the type you thought it should have been but that's a different story).  Do you have a reproducer?

Reply:
It actually does work. Running `input->Show(0)` shows the correct values. It's the vscode root file viewer that has a bug on it. Refactored to have all of them to have only the refs and let ROOT resolve the type automatically (just like the other ones). Thanks for making me double check this.